documentation/secure_storage.md

```diff
@@ -0,0 +1,136 @@
+# Secure Storage In OP-TEE
+
+## Background
+
+Secure Storage (SST) in OP-TEE is implemented according to what has been defined
+in GloblaPlatform’s TEE Internal API specification (here called Trusted
+Storage). This specification mandates that it should be possible to store
+general-purpose data and key material that guarantees confidentiality and
+integrity of the data stored and the atomicity of the operations that modifies
+the storage (atomicity here means that either the entire operation completes
+successfully or no write is done).
+
+## Overview
+
+![Secure Storage System Architecture](images/secure_storage/secure_storage_system_architecture.png
```

Comment:
I don't think we should use image files like this. Of course they look neat, but they are difficult to maintain, all the more since the source files for the pictures are not part of the PR. With binary pictures you cannot easily see what's changed from version to version. And IMO they do not bring much more value than simple text drawings.